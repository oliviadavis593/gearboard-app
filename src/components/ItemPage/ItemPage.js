import React, { Component } from 'react';
import GearContext from '../../GearContext'
import EditItem from '../EditItem/EditItem'
import '../../styles/ItemPage.css'


export default class ItemPage extends Component {

    
    static defaultProps =  {
        history: {
            goBack: () => {}
        },
        match: {
            params: {}
        }
    }

    static contextType = GearContext; 

    render() {
        const items = this.context.items; 
        const { item_id } = this.props.match.params
        const item = items.find(item => item.id.toString() === item_id.toString()) || {}

        return(
                <section className='gb-item-page__container'>
                <header>
                    <h1>Gear Item</h1>
                </header>
                <div className='gb-item-info__container'>
                    <div className='item__rating'>
                        <h2>Rating</h2> 
                        <div className='gb-rating'>
                            {item.rating}
                        </div>
                    </div>
                    <div className='item__features'>
                        <h2>Features</h2> 
                        <div className='gb-features'>
                            {item.features}
                        </div>
                    </div>
                    <div className='item__comments'>
                        <h2>Comments</h2> 
                        <div className='gb-comments'>
                            {item.comments}
                        </div>
                    </div>
                    </div>
                <div className='gb-edit-btns__container'>
                    <div>
                        <EditItem 
                        id={item_id}
                        />
                    </div>
                <button
                    tag='button'
                    role='link'
                    className='gb-edit__goback-button'
                    onClick={() => this.props.history.goBack()} 
                >
                    Back
                </button>
                </div>
            </section>
        )
    }
}