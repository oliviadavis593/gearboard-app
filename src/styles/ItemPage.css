h1 {
    text-align: center; 
}

.gb-item-info__container {
    border: 1px solid #eee;
    height: 100%; 
    width: 300px; 
    margin: 30px auto; 
    background-color: #eee; 
    box-shadow: 0 2px 4px 0 rgba(0,0,0,0.4);
}

.gb-rating,
.gb-features,
.gb-comments {
    margin: 0 auto;  
    width: 250px; 
    text-align: center; 
}

.gb-edit-btns__container {
    display: flex; 
    justify-content: center;
}

.gb-edit__goback-button {
    text-decoration: none;
    color: #000;
    font-family: 'Merriweather', serif;
    background-color: #e67e22;
    width: 100px; 
    height: 40px; 
    border: none; 
    border-radius: 20px; 
    font-size: 15px;  
    margin: 20px  auto; 
}

/*========Media Queries============*/
@media only screen and (min-width: 600px) {
   h1 {
       font-size: 35px; 
   }
}

@media only screen and (min-width: 1400px) {
    .gb-item-info__container {
        width: 350px;
        font-size: 20px; 
    }


}