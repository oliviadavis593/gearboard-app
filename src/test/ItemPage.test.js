import React from 'react';
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import ItemPage from '../components/ItemPage/ItemPage'


describe('Footer component', () => {
    it(`renders without crashing`, () => {
        const div = document.createElement('div')
        ReactDOM.render(<ItemPage />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<ItemPage rating='ðŸŽ¸' features='features' comments='comments'/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})