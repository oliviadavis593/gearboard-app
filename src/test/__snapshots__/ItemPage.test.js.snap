// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders the UI as expected 1`] = `
<section
  className="gb-item-page__container"
>
  <header>
    <h1>
      Gear Item
    </h1>
  </header>
  <div
    className="gb-item-info__container"
  >
    <div
      className="item__rating"
    >
      <h2>
        Rating
      </h2>
      <div
        className="gb-rating"
      />
    </div>
    <div
      className="item__features"
    >
      <h2>
        Features
      </h2>
      <div
        className="gb-features"
      />
    </div>
    <div
      className="item__comments"
    >
      <h2>
        Comments
      </h2>
      <div
        className="gb-comments"
      />
    </div>
    <div>
      <a
        className="gb-edit__route"
        href="/edit/undefined"
        onClick={[Function]}
      >
        Edit Item
      </a>
    </div>
  </div>
  <div>
    <button
      className="gb-edit__goback-button"
      onClick={[Function]}
      role="link"
      tag="button"
    >
      Back
    </button>
  </div>
</section>
`;
